app/models/event.rb

```diff
@@ -70,7 +70,11 @@ def add_test_teams
   end
 
   def generate_schedule
-    calculate_round_robin
+    if game_mode == 0
+      calculate_round_robin
+    elsif game_mode == 1
+      calculate_double_round_robin
```

Comment:
it would be neat, if you could use the actual enum here for more readability. Also in the new version of the dev this code is in the league.rb instead of the event.rb as you'll see when you merge.